### ISSUE TITLE
Regression in Forced Alignment Accuracy Post-Commit efcd95b - Incorrect Context Removal in generate_emissions

### ISSUE THREAD

**Issue by leeseungchae** [MahmoudAshraf97/ctc-forced-aligner#57]
**Description:**

Forced alignment accuracy has regressed after commit efcd95bd30958a95d6ba4375e760cb1550bf633a.  

This regression was observed during **Korean language testing. .**..

Analysis points to the `generate_emissions` function, 

where context removal is now incorrectly performed after flattening, degrading accuracy, 

especially for long audio.

**Problem:**

The current `generate_emissions` function (post- efcd95bd30958a95d6ba4375e760cb1550bf633a ) flattens emissions before context removal.  

This reverses the correct order and leads to inaccurate context trimming in window-based processing.

**Code Snippets:**

**Current (Incorrect - post- efcd95bd30958a95d6ba4375e760cb1550bf633a ):**

```python
emissions = torch.cat(emissions_arr, dim=0).flatten(0, 1)
if context > 0:
      emissions = emissions[
            time_to_frame(context_length) : -time_to_frame(context_length) + 1,
        ]  # removing the context  ```
```

**Correct (Pre- efcd95bd30958a95d6ba4375e760cb1550bf633a ):**

```python 
emissions = torch.cat(emissions_arr, dim=0)[
        :,
        time_to_frame(context_length) : -time_to_frame(context_length) + 1,
    ]  # removing the context
emissions = emissions.flatten(0, 1)
```

**Issue Explanation:** 

Incorrectly removing context after flattening leads to inaccurate trimming at window boundaries. 

Instead of removing only the added context from each window, the process affects the underlying audio representation at window seams. 

This degrades alignment accuracy, especially for longer audio processed in windows.

**Steps to Reproduce:**

Compare forced alignment accuracy on long audio files using code before and after commit efcd95b.  

The regression is evident in reduced accuracy and synchronization in the post-efcd95b version.  

Increasing batch_size can amplify the effect.

**Impact:**

Significantly reduced forced alignment accuracy, especially for longer audio, 

impacting core functionality. Regression introduced in commit efcd95bd30958a95d6ba4375e760cb1550bf633a.

Proposed Solution:

Revert the context removal logic in generate_emissions to the pre- efcd95bd30958a95d6ba4375e760cb1550bf633a implementation. 

Ensure context removal happens before flattening.

**Urgency:**

High - Critical regression impacting core accuracy.

**Comment by MahmoudAshraf97** [comment]
Thanks for the detailed report I'll look into it

**Comment by MahmoudAshraf97** [comment]
@leeseungchae can you check if #58 solves the issue?

**Comment by leeseungchae** [comment]
It seems like the sync issue is resolved after #58
